playground/backend/internal/executors/java_helper_test.go

```diff
@@ -0,0 +1,177 @@
+	err := javaExecutor.Validate()
+	if err != nil {
+		t.Fatalf(`TestValidateJavaFile error: %v `, err)
+	}
+}
+
+func TestCompileJavaFile(t *testing.T) {
+	err := javaExecutor.Compile()
+	if err != nil {
+		t.Fatalf("TestCompileJavaFile: Unexpexted error at compiliation: %s ", err.Error())
+	}
+}
+
+func TestRunJavaFile(t *testing.T) {
+	className := "HelloWorld"
+	expected := "Hello World!\n"
+	out, err := javaExecutor.Run(className)
+	if expected != out || err != nil {
+		t.Fatalf(`TestRunJavaFile: '%q, %v' doesn't match for '%#q', nil`, out, err, expected)
+	}
+}
+
+func TestGetJavaPreparation(t *testing.T) {
+	prep1 := preparationWithArgs{
+		prepare: removePublicClassModification,
+	}
+	prep2 := preparationWithArgs{
+		prepare: removeAdditionalPackage,
+	}
+	prep := []preparationWithArgs{prep1, prep2}
+	tests := []struct {
+		name string
+		want *[]preparationWithArgs
+	}{
+		{
+			name: "GetJavaPreparation",
+			want: &prep,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := GetJavaPreparation(); !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("GetJavaPreparation() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func Test_removeAdditionalPackage(t *testing.T) {
+	javaFS, _ := fs_tool.NewLifeCycle(pb.Sdk_SDK_JAVA, pipelineId)
+	type args struct {
+		filePath string
+		args     []interface{}
+	}
+	tests := []struct {
+		name    string
+		args    args
+		wantErr bool
+	}{
+		{
+			name: "original file doesn't exist",
+			args: args{
+				filePath: "someFile.java",
+				args:     make([]interface{}, 0),
+			},
+			wantErr: true,
+		},
+		{
+			name: "original file exists",
+			args: args{
+				filePath: javaFS.GetAbsoluteExecutableFilePath(),
+				args:     make([]interface{}, 0),
+			},
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if err := removeAdditionalPackage(tt.args.filePath, tt.args.args...); (err != nil) != tt.wantErr {
+				t.Errorf("removeAdditionalPackage() error = %v, wantErr %v", err, tt.wantErr)
+			}
+		})
+	}
+}
+
+func Test_removePublicClassModification(t *testing.T) {
+	javaFS, _ := fs_tool.NewLifeCycle(pb.Sdk_SDK_JAVA, pipelineId)
+	type args struct {
+		filePath string
+		args     []interface{}
+	}
+	tests := []struct {
+		name    string
+		args    args
+		wantErr bool
+	}{
+		{
+			name: "original file doesn't exist",
+			args: args{
+				filePath: "someFile.java",
+				args:     make([]interface{}, 0),
+			},
+			wantErr: true,
+		},
+		{
+			name: "original file exists",
+			args: args{
+				filePath: javaFS.GetAbsoluteExecutableFilePath(),
+				args:     make([]interface{}, 0),
+			},
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if err := removePublicClassModification(tt.args.filePath, tt.args.args...); (err != nil) != tt.wantErr {
+				t.Errorf("removeAdditionalPackage() error = %v, wantErr %v", err, tt.wantErr)
+			}
+		})
+	}
+}
```

Comment:
Same as with package – should we check that we actually remove the public class modification that we want to?